### PR TITLE
Feat: Add new step 2 - Method 1

### PR DESCRIPTION
**Pull request by PoulainMaxime**
Screenshot 2022-09-01 at 15 59 06

- [x] Unit test

### CODE REVIEW

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/delivery-time-and-rates.vue [hunk 1]
```diff
@@ -70,8 +210,15 @@ export default {
...
+    getRate(value) {
+      this.rateChosen = value;
+    },
     previousStep() {
-      localStorage.setItem('productFeed-deliveryDetails', JSON.stringify(this.carriers));
+      if (this.getShippingValueSetup === ShippingSetupOption.IMPORT) {
+        localStorage.setItem('productFeed-customCarrier', JSON.stringify(this.customCarrier));
+      } else {
+        localStorage.setItem('productFeed-deliveryDetails', JSON.stringify(this.carriers));
+      }
```
Comment (Quetzacoalt91) [inline _dev/src/components/product-feed/settings/delivery-time-and-rates/delivery-time-and-rates.vue:221]: You mpean to save the current changes when going to the previous step? This should not happen, we even warn that changes will be lost

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        });
+
+        return currencyFormatted.replace(/[ .,0]*/, '');
+      } catch (error) {
+        const currency = symbols.find((c) => c.currency === currentCurrency);
+
+        return currency ? currency.symbol : '';
+      }
+    },
+  },
```
Comment (Quetzacoalt91) [inline _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue:310]: Duplicated code from _dev/src/components/campaign-creation/campaign-creation.vue. Time to extract it

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+                      <span
+                        class="font-weight-600"
+                      >
+                        {{ $t('productFeedSettings.deliveryTimeAndRates.estimateStep.shippingRate.rate') }}
+                      </span>
+                      <b-button
+                        class="p-0 ml-1"
+                        variant="text"
+                        v-b-tooltip:psxMktgWithGoogleApp
+                        :title="''"
```
Comment (Quetzacoalt91) [inline _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue:198]: What is this line about?

File: _dev/src/assets/json/translations/en/ui.json [hunk 4]
```diff
@@ -258,6 +258,30 @@
...
+        "rateDesc": "I can estimate the same shipping rate and delivery time for all countries",
+        "customTitle": "Custom rates by country",
+        "customDesc": "I'd like to set different rates and times for each country"
+      },
+      "estimateStep": {
+        "error": "Fill in the required information",
+        "carrierName": "Carrier name",
+        "storeOffersTitle": "My store offers",
+        "storeOffers": {
+          "freeShippingRateForAllProducts": "Flat shipping rate for all products",
```
Comment (Quetzacoalt91) [inline _dev/src/assets/json/translations/en/ui.json:275]: Not sure about the name. Should be

```suggestion
          "flatShippingRateForAllProducts": "Flat shipping rate for all products",
```
?

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+      if (this.carrier.carrierName.length <= 90
+        && this.carrier.carrierName.length > 0
+      ) {
+        return true;
+      }
+
+      return false;
+    },
+    validateRadio(): boolean|null {
+      if (this.carrier.offerChosen) {
```
Comment (Quetzacoalt91) [inline _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue:286]: Did you mean to check if the offer is not chosen yet?

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue (hunk 3 above)
Comment (PoulainMaxime) [reply _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue:198]: Is for the tooltip text but I don't have it

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue (hunk 5 above)
Comment (PoulainMaxime) [reply _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue:286]: Yes i'm checking if value is already set

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/delivery-time-and-rates.vue [hunk 6]
```diff
@@ -4,41 +4,184 @@
...
     shippingSettingsHeaderList() {
       return Object.values(ShippingSettingsHeaderType);
     },
+    countries(): string[] {
+      return this.$store.getters['productFeed/GET_TARGET_COUNTRIES'];
+    },
+    hasMultipleCountries(): boolean {
+      return this.countries.length > 1;
+    },
+    convertToCountryName(): string[] {
```
Comment (Quetzacoalt91) [inline _dev/src/components/product-feed/settings/delivery-time-and-rates/delivery-time-and-rates.vue:179]:
```suggestion
    countriesNames(): string[] {
```

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+      if (!this.carrier.carrierName?.length) {
+        return null;
+      }
+      if (this.carrier.carrierName.length <= 90
+        && this.carrier.carrierName.length > 0
+      ) {
+        return true;
+      }
+
+      return false;
```
Comment (Quetzacoalt91) [inline _dev/src/components/product-feed/settings/delivery-time-and-rates/create-custom-carrier.vue:283]: Duplicate of the method of the same name in _dev/src/providers/shipping-rate-provider.ts?

File: _dev/src/components/product-feed/settings/delivery-time-and-rates/delivery-time-and-rates.vue [hunk 8]
```diff
@@ -75,8 +109,15 @@ export default {
...
+    getRate(value) {
+      this.rateChosen = value;
+    },
     previousStep() {
-      localStorage.setItem('productFeed-deliveryDetails', JSON.stringify(this.carriers));
+      if (this.getShippingValueSetup === ShippingSetupOption.IMPORT) {
+        localStorage.setItem('productFeed-customCarrier', JSON.stringify(this.customCarrier));
+      } else {
+        localStorage.setItem('productFeed-deliveryDetails', JSON.stringify(this.carriers));
+      }
```
Comment (Quetzacoalt91) [inline _dev/src/components/product-feed/settings/delivery-time-and-rates/delivery-time-and-rates.vue:120]:
```suggestion
```

We shouldn't save anything in the locastorage if we go back

**Review by Quetzacoalt91 — APPROVED** [review]

**Pull request merged**